gateway/ln-gateway/src/lib.rs

```diff
@@ -197,6 +199,9 @@ pub struct Gateway {
     /// The gateway's federation manager.
     federation_manager: Arc<RwLock<FederationManager>>,
 
+    /// The gateway's authentication manager
+    auth_manager: Arc<Mutex<AuthManager>>,
```

Comment:
Why does the auth manager need to be mutated at all? Can we remove the `Acr<Mutex<_>>`?